### DISCUSSION TITLE
Redefine certain behaviors (like 0^0) or "catch" operations like that

### DISCUSSION THREAD

**Discussion by NullDev** [josdejong/mathjs#3171] [Q&A] [answered]
Hey, it's me again with yet another weird question 👋 
For another project I need rather strict mathematical definitions. For example, currently 0^0 results in 1. 
But for the sake of my project I'd rather keep it as undefined. Now a simple solution would be just doing e.g. 

```js
if (String(inputExpression).replaceAll(" ", "").includes("0^0")){
    throw new Error("0^0 is undefined");
}
```

but that fails for a.) different notations like `0^(0)` and b.) not fully computed expressions like `0^(1-1)`.
Is there a clean way to either redefine the result of 0^0 or to catch this exact operation somehow (preferably without forking the lib)?

**Comment by josdejong** [comment, marked as answer]
> Hey, it's me again with yet another weird question 👋

😂 hi!

You'll have to replace the implementation of `math.pow` with your own version, that handles the case of `0^0` like you want.

For example:

```js
const originalPow = math.pow

function customPow(x, y) {
  if (math.isZero(x) && math.isZero(y)) {
    return undefined
  }
  
  return originalPow(x, y)
}

math.import({ pow: customPow }, { override: true })

console.log('result:', math.evaluate('0^0')) // undefined
```

**Reply by NullDev** [reply]
This is perfect, thank you (once again)! 😄
